libnetdbpf: Reduce error logspam

Terminator-J opened the pull request:
Somewhere between Android and netd, the localhost loopback (interface index #1) is either not getting or is losing the name that BPF expects, causing any services using 127.0.0.1 (like when using a hostfile-based ad blocker with root) to trip the traffic threshold for bytes on an unnamed interface and spew "Unknown name for ifindex 1 with more than ____ bytes of traffic" errors constantly.

Changed log level to debug for now, because the "error" is that something else is broken.

Uldiniad committed:
system/netd: Squash of netd uid bandwidth restriction commits

Author: Uldiniad <olivercscott@gmail.com>
Date:   Thu Mar 1 08:54:00 2018 -0500

    [3/3] NetD : Allow passing in interface names for wifi/data app restriction

    CYAN-3976
    CRACKLING-834

    This is a simplification and adpatation for oreo of:
    https://review.lineageos.org/#/c/LineageOS/android_system_netd/+/144246/
    by wangjing <wangjing@codeaurora.org>

    Change-Id: I085434d70dfe00e9c27b821661fff5076d57e930

----

Author: Uldiniad <olivercscott@gmail.com>
Date:   Tue Oct 30 23:59:08 2018 +0000

    NetD : Allow passing in interface names for vpn app restriction

    Change-Id: I7cb0c895db60dcf6b4fe59732ec6c0ec3c212a04

----

Author: Sam Mortimer <sam@mortimer.me.uk>
Date:   Thu Aug 29 16:48:55 2019 -0700

    netd: Consolidate restrict apps methods

    * These are lineage additions (that originated
      from caf).

    * addRestrictAppsOnData, addRestrictAppsOnVpn and
      addRestrictAppsOnWlan all do a similar thing
      (fw/b passes different interface arguments).

    * Consolidate into addRestrictAppsOnInterface
      (and removeRestrictAppsOnInterface)

    * Requires corresponding fw/b services change.

    Change-Id: Ic780e943583e616c137c42aa4a72f7af63d02dce

----

Author: Sam Mortimer <sam@mortimer.me.uk>
Date: Fri Apr 10 19:19:47 2020 -0400

    system/netd: Add BPF support for lineage network restrictions

    Required for kernels 4.9+

    Change-Id: I369c95b33742f8b3f95764ab43fbfb20dd4ec0e5

Change-Id: I3d1c0347ecaf343cd93a4fbb0c0e1a64d010b0d7

Commit:
netd: restrict DNS names

Do not accept WPAD or ISATAP from DHCP as names to public in DNS.

Change-Id: Ic88651bcf4b36205908e65fef41ea08cdca5e0b9
CRs-fixed: 2285921
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>

Lukas0610 committed:
netd: Allow devices to force-add directly-connected routes

Most sane devices have directly-connected routes properly
added by ConnectivityService, but a number of Samsung RIL
libraries seem to be doing "special" things, the end result
of which is IPv4 mobile data is non-functional due to lack
of a default route.

Affected devices have a log entry similar to the below:
  Error adding route 0.0.0.0/0 -> xxx.xxx.xxx.xxx rmnet0 to table 1002: Network is unreachable

This reverts, with guards, commit 92e8f96e43320efd5183d7452fb90883fd96415e.

Change-Id: I6a653d19c1c72c136e78d8d841e87d3c166698ea

Commit:
Netd: Add support for quotaReachedAlert event

Handle quotaReachedAlert event from kernel and
notify the limit alert to NMS.

Test : Quota related test cases

Change-Id: I90d2c5769a975bcaffbb2509d59f392811075cc2
CRs-Fixed: 2485528
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>

sam3000 committed:
system/netd: Add support for per app network isolation

* Add a new firewall chain fw_isolated that blocks all network access.

* Legacy ipchains and bpf are both supported.

Change-Id: Ie3c8662ebaf9ee90d7207b427bb9a9b3a2e86cd0

rpavanqcom committed:
Check for interface name validity

Support interface names with '.' character.

Test: builds, boots
Change-Id: I0490df2f8b3cda1453cf9ed2d3290c2983240f5e

chirayudesai committed:
bw: Always apply penalty box

* Why should only costly interfaces have all the fun

Change-Id: I634c4c8a8e83dd55ac287efcebda1980571c8772
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>

Uldiniad committed:
system/netd: add missing override keyword

Change-Id: I5ce309f47d9b37dd359d39419fe185feacad7e13

Terminator-J commented:
This didn't come out for enchilada/fajita until recently when luk1337 pushed an update to oneplus_sdm845-common device tree  (https://github.com/LineageOS/android_device_oneplus_sdm845-common/commit/2fa620b44f3b4500c93035dfaaab164009e338fd) that set eBPF support level and suddenly I'm getting this ridiculous amount of log error spam. 

I can try reverting it and see what starts complaining, which would be useful for tracking down the root cause, but for now let's make the symptoms less obnoxious because nothing is actually failing to work -- not everything needs to be an 'error' instead of a debug or warning message, hey?

gwolf2u commented:
I would rather find root cause of this than just changing log type
just my two cents

Terminator-J committed:
Clat eBPF is always enabled (and required)

Android S requires devices to support eBPF, additionally ClatEbpfMaybe
was only used to reduce some errors to informational logging.

This is no longer worth it, as launch Q devices will never trigger
this, and if anyone decides to upgrade a 4.9+ Pie device they can
simply enable the relevant options (such devices are likely already
running Q era kernels anyway...).

ie. CONFIG_NET_SCH_INGRESS and CONFIG_NET_CLS_BPF are required

Test: builds, atest, TreeHugger
Bug: 167500195
Signed-off-by: Maciej Żenczykowski <maze@google.com>
Change-Id: I06dc92d5fd477886aa50e8751cb13710d6b98427
Signed-off-by: DennySPb <dennyspb@gmail.com>

gwolf2u commented:
closed as not preferred way of doing it
